app/grandchallenge/reader_studies/views.py

```diff
@@ -822,35 +830,73 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class AnswersRemove(
-    LoginRequiredMixin,
-    ObjectPermissionRequiredMixin,
-    SuccessMessageMixin,
-    FormView,
-):
-    template_name = "reader_studies/readerstudy_user_groups_form.html"
-    form_class = AnswersRemoveForm
-    success_message = "Answers removed"
+class AnswerBatchDelete(LoginRequiredMixin, DeleteView):
     permission_required = (
-        f"{ReaderStudy._meta.app_label}.change_{ReaderStudy._meta.model_name}"
+        f"{Answer._meta.app_label}.delete_{Answer._meta.model_name}"
     )
     raise_exception = True
+    success_message = "Answers removed"
 
-    def get_permission_object(self):
-        return self.reader_study
+    def check_permissions(self, request):
+        permission_objects = self.get_queryset()
+        forbidden = any(
+            not request.user.has_perm(self.permission_required, obj)
+            for obj in permission_objects
+        )
+        if forbidden:
+            raise PermissionDenied()
+        return permission_objects
+
+    def get_queryset(self):
+        raise NotImplementedError
+
+    def delete(self, request, *args, **kwargs):
+        objects = self.check_permissions(request)
+        objects.delete()
+        messages.add_message(request, messages.SUCCESS, self.success_message)
+        return HttpResponse(
+            self.get_success_url(),
+            headers={
+                "HX-Redirect": self.get_success_url(),
+                "HX-Refresh": True,
+            },
+        )
 
     @property
     def reader_study(self):
         return get_object_or_404(ReaderStudy, slug=self.kwargs["slug"])
 
-    def form_valid(self, form):
-        form.remove_answers(reader_study=self.reader_study)
-        return super().form_valid(form)
-
     def get_success_url(self):
         return self.reader_study.get_absolute_url()
 
 
+class AnswersRemoveForUser(AnswerBatchDelete):
+    def get_queryset(self):
+        data = QueryDict(self.request.body)
+        user = data["user"]
```

Comment:
This is a strange construction and results in strange API calls in the tests - `data=f"user={r1.id}"`. Shouldn't a form with validation of the user field, or using a URL with a capture value for the username (`"/remove-answers/<str:username>/"`, `creator__username=...`) be nicer?

Reply:
I'll include it in the url, as html5 does not support the delete method for forms.